app/models/user.rb

```diff
@@ -33,4 +33,25 @@ def is_member_or_admin?
     admin? || is_member?
   end
 
+  def is_in_company_numbered?(company_num)
+    is_member? && !(companies.detect{|c| c.company_number == company_num}).nil?
+  end
+
+  def companies
```

Comment:
This method could use some sliming down..
I could be wrong but the following should in principle work the same
```ruby
def companies
  if admin?
    Company.all
  elsif is_member_or_admin? && has_membership_application?
    cos = membership_applications.reload.map(&:company).compact
    cos.uniq(&:company_number)
  else
    [] # no_companies
  end
end
```